src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/JSON2SolutionGraphImpl.java

```diff
@@ -0,0 +1,233 @@
+ */
+public class JSON2SolutionGraphImpl implements JSON2SolutionGraph {
+
+    final protected GraphQL2RDFConfiguration config;
+    final protected GraphQLEndpoint endpoint;
+
+    public JSON2SolutionGraphImpl(final GraphQL2RDFConfiguration config, 
+                                     final GraphQLEndpoint endpoint){
+        this.config = config;
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    public Model translateJSON( final JSONResponse jsonResponse) throws ParseException {
+
+        final JsonObject outerJson = jsonResponse.getJsonObject();
+        final Model solutionGraph = ModelFactory.createDefaultModel();
+        
+        // Maps a GraphQL type then its ID to a blank node representing the GraphQL object
+        final Map<BlankNodeID,Resource> blankNodes = new HashMap<>();
+
+        // Verify that the JSONResponse contains correct data throw error
+        if(!outerJson.hasKey("data")){
+            throw new ParseException("The json response did not contain a data key.");
+        }
+
+        final JsonObject json = outerJson.getObj("data");
+
+        // Parse through each individual entrypoint in the JSON 
+        // (the return value from parseJSON isn't necessary on the original call)
+        for(final String entrypoint : json.keySet()){
+
+            final JsonValue e = json.get(entrypoint);
+
+            if(e.isArray()){
+                for(final JsonValue v : e.getAsArray()){
+                    if(v.isObject()){
+                        parseJSON(v.getAsObject(), blankNodes, solutionGraph);
+                    } 
+                }
+            }
+            else if(e.isObject()){
+                parseJSON(e.getAsObject(), blankNodes, solutionGraph);
+            }
+        }
+
+        return solutionGraph;
+    }
+
+    /**
+     * Parses through a JsonObject recursively and expands @param blankNodes and @param solutionGraph.
+     * @return a Resource in order to create a triple with the parent and the child resource.
+     * @throws ParseException if the json object does not contain a valid "id" key.
+     */
+    protected Resource parseJSON(final JsonObject root, 
+                                 final Map<BlankNodeID,Resource> blankNodes, 
+                                 final Model solutionGraph) throws ParseException {
+
+        // Get the key that represent the current json objects id
+        final String idKey = root.keySet()
+            .stream()
+            .filter(s -> s.startsWith(config.getJsonIDKeyPrefix()))
+            .findAny()
+            .orElse(null);
+
+        if(idKey == null){
+            throw new ParseException("JsonObject root did not provide a valid id key.");
+        }
+
+        // Initialize the necessary data needed for parsing the current json object
+        final String graphqlID = root.getString(idKey);
+        root.keySet().remove(idKey);
+        final String graphqlType = removeJsonKeyPrefix(idKey);
```

Comment:
Maybe this should be managed by the configuration? Alternatively, this can be a method on a utility class.